inc/class-core-sitemaps.php

```diff
@@ -110,11 +111,11 @@ public function register_rewrites() {
 		add_rewrite_rule( '^wp-sitemap-index\.xsl$', 'index.php?sitemap-stylesheet=index', 'top' );
 
 		// Register routes for providers.
-		$providers = core_sitemaps_get_sitemaps();
-
-		foreach ( $providers as $provider ) {
-			add_rewrite_rule( $provider->route, $provider->rewrite_query(), 'top' );
-		}
+		add_rewrite_rule(
+			'^wp-sitemap-([A-z]+)-?([A-z]+)-?([0-9]+)?\.xml$',
+			'index.php?sitemap=$matches[1]&sitemap-sub-type=$matches[2]&paged=$matches[3]',
```

Comment:
What about also renaming the query param to `wp-sitemap` with this MR?

Reply:
Query vars are not typically prefixed in WP. Don't think we should make an exception here.